### PR TITLE
refactor(config): remove `GetConfig` and `config` global

### PR DESCRIPTION
**Pull request by ThinkChaos**
First step  towards completely removing the `config` global: remove `GetConfig`.  
This is done by moving a couple options under `Upstreams` and embedding `Upstreams` into `UpstreamGroup` so we can access those options from resolvers easily.

~I'll completely remove the global in a followup PR.~ It was easier than I thought, done!

---

[Rendered docs](https://thinkchaos.github.io/blocky/refactor-private-config-global/configuration/#upstreams-configuration)

### CODE REVIEW

**Comment by ThinkChaos** [comment]
Rebased it to not have a merge commit and give the e2e tests a retry

**Review by kwitsch — APPROVED** [review]
Only a minor comment that could be resolved later on.

👍

File: server/server.go [hunk 1]
```diff
@@ -56,8 +56,8 @@ func logger() *logrus.Entry {
 	return log.PrefixedLog("server")
 }
 
-func minTLSVersion() uint16 {
-	minTLSVer := config.GetConfig().MinTLSServeVer
+func minTLSVersion(cfg *config.Config) uint16 {
+	minTLSVer := cfg.MinTLSServeVer
```
[the remark this answers is not in the document]
Comment (ThinkChaos) [reply server/server.go:60]: Good idea and seems simple enough, I'll add it to this PR :)

**Comment by kwitsch** [comment]
👍

**Pull request merged**